config/networking/first-party-jwt-istio-for-kind.yml

```diff
@@ -0,0 +1,57 @@
+#@ load("@ytt:data", "data")
+#@ load("@ytt:json", "json")
+#@ load("@ytt:overlay", "overlay")
+#@ load("@ytt:yaml", "yaml")
+
+#@ if data.values.patch_for_kind:
```

Comment:
Hmm, it sounds like this config-optional file was not as kind-specific as RelEng found that they need this for PKS/vsphere [vmware slack thread](https://vmware.slack.com/archives/C3LUD2L04/p1596764860097900?thread_ts=1596762799.097400&cid=C3LUD2L04)

Reply:
maybe that's solved by adding an `or` here?

Reply:
It sounds like we are running into more of these sorts of things, and that we generally don't want to tie them directly to a specific IaaS.

Reply:
I would favor [feature detection](https://en.wikipedia.org/wiki/Feature_detection_(web_development)) over IaaS/version. This is something I feel we need to discuss in a wider setting if possible. @jamespollard8 @davewalter @nhsieh

Reply:
I agree that feature detection would be preferable for such cases. As a platform operator I just want the right thing to happen, not find out the right set of config knobs.

Reply:
e.g. for this particular case:

To determine if your cluster supports third party tokens, look for the TokenRequest API:

```console
$ kubectl get --raw /api/v1 | jq '.resources[] | select(.name | index("serviceaccounts/token"))'
{
    "name": "serviceaccounts/token",
    "singularName": "",
    "namespaced": true,
    "group": "authentication.k8s.io",
    "version": "v1",
    "kind": "TokenRequest",
    "verbs": [
        "create"
    ]
}
```